spread-shellcheck

```diff
@@ -160,16 +161,20 @@ def checkfile(path):
 
     if path.endswith('spread.yaml') and 'suites' in data:
         # check suites
-        for suite in data['suites'].keys():
-            for section in SECTIONS:
-                if section not in data['suites'][suite]:
-                    continue
+        with ThreadPoolExecutor() as executor:
```

Comment:
maybe we should set `max_workers=N` like we do for the other one

Reply:
I'd remove both options frankly, the default is correct and we don't need to slow this down.

Reply:
I think landing as is in fine but we should do a followup that make it consistent again with checkpath and respecting --max-procs (or removing this option).

 I'm not super sure if the default (in the released versions of ubuntu) is really great, it says:
"""


Changed in version 3.5: If max_workers is None or not given, it will default to the number of processors on the machine, multiplied by 5, assuming that ThreadPoolExecutor is often used to overlap I/O instead of CPU work and the number of workers should be higher than the number of workers for ProcessPoolExecutor.
"""
(we have 3.5 in xenial and 3.8 in focal/groovy). This was changed again in 3.9.

Reply:
Hm I suppose the logistics of passing max_workers can be a little annoying. Maybe it's time to add a class and make respective functions methods of that class?